linux/baraction.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/sh
+# Example Bar Action Script for Linux.
+# Requires: acpi, iostat.
+# Tested on: Debian 10, Fedora 31.
+#
+
+print_date() {
+	# The date is printed to the status bar by default.
+	# To print the date through this script, set clock_enabled to 0
+	# in spectrwm.conf.  Uncomment "print_date" below.
+	FORMAT="%a %b %d %R %Z %Y"
+	DATE=`date "+${FORMAT}"`
+	echo -n "${DATE}     "
+}
+
+print_mem() {
+	MEM=`/usr/bin/free -m | grep ^Mem: | sed -E 's/ +/ /g' | cut -d ' ' -f4`
+	echo -n "Free mem: ${MEM}M  "
+}
+
+_print_cpu() {
+	printf "CPU: %3d%% User %3d%% Nice %3d%% Sys %3d%% Idle  " $1 $2 $3 $6
+}
+
+print_cpu() {
+	OUT=""
+	# Remove the decimal part from all the percentages.
+	while [ "${1}x" != "x" ]; do
+		OUT="$OUT `echo "${1}" | cut -d '.' -f1`"
+		shift;
+	done
+	_print_cpu $OUT
+}
+
+print_cpuspeed() {
+	CPU_SPEED=`/usr/bin/lscpu | grep '^CPU MHz:' | sed -E 's/ +/ /g' | cut -d ' ' -f3 | cut -d '.' -f1`
+	printf "CPU speed: %4d MHz  " $CPU_SPEED
+}
+
+print_bat() {
+	AC_STATUS="$3"
+	BAT_STATUS="$6"
+	# Most battery statuses fit into a single word, except "Not charging"
+	# for which we need to have special handling.
+	if [ "$BAT_STATUS" = "Not" ]; then
+		shift
```

Comment:
When battery status is "Not Charging",  $BAT_STATUS is still "Not" after the *shift* resulting in "(battery unknown)" instead of "(battery $BAT_LEVEL)".

Reply:
Good catch - I made some last-minutes tweaks and messed up the logic in the process. I'll take care of it in the next version.